src/dbnode/storage/bootstrap/bootstrapper/commitlog/source.go

```diff
@@ -518,36 +587,13 @@ func (s *commitLogSource) Read(
 	shouldReturnUnfulfilled, err := s.shouldReturnUnfulfilled(
 		workers, encounteredCorruptData, initialTopologyState)
 	if err != nil {
-		return bootstrap.NamespaceResults{}, err
+		return commitLogResult{}, err
 	}
-
-	bootstrapResult := bootstrap.NamespaceResults{
-		Results: bootstrap.NewNamespaceResultsMap(bootstrap.NamespaceResultsMapOptions{}),
-	}
-	for _, ns := range namespaceResults {
-		id := ns.namespace.Metadata.ID()
-		dataResult := result.NewDataBootstrapResult()
-		if shouldReturnUnfulfilled {
-			shardTimeRanges := ns.namespace.DataRunOptions.ShardTimeRanges
-			dataResult = shardTimeRanges.ToUnfulfilledDataResult()
-		}
-		var indexResult result.IndexBootstrapResult
-		if ns.namespace.Metadata.Options().IndexOptions().Enabled() {
-			indexResult = result.NewIndexBootstrapResult()
-			if shouldReturnUnfulfilled {
-				shardTimeRanges := ns.namespace.IndexRunOptions.ShardTimeRanges
-				indexResult = shardTimeRanges.ToUnfulfilledIndexResult()
-			}
-		}
-		bootstrapResult.Results.Set(id, bootstrap.NamespaceResult{
-			Metadata:    ns.namespace.Metadata,
-			Shards:      ns.namespace.Shards,
-			DataResult:  dataResult,
-			IndexResult: indexResult,
-		})
+	s.commitLogResult = commitLogResult{
```

Comment:
nit:  It's a little awkward to set it here and use the set value as the return value. We also use the return value after the function call. Maybe it's better to either have no return or return the value and let the caller do the setting?

Reply:
yea I don't have a strong opinion about this since it's all private local state anyways. I'll just update it so the main function controls the caching logic and decides if it should actually call `readCommitLog`